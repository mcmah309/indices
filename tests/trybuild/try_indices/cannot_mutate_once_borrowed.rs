use indices::try_indices;

fn main() {
    let mut slice = [0, 1, 2, 3, 4];
    let (four, _one, _two) = try_indices!(&mut slice, 4, 1, 2).unwrap();
    slice[4] *= 2;
    *four *= 2;
    slice[4] *= 2;
    *four *= 2;
    println!("{four}");
}