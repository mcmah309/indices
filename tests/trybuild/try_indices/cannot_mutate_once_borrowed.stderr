error[E0503]: cannot use `slice` because it was mutably borrowed
 --> tests/trybuild/try_indices/cannot_mutate_once_borrowed.rs:6:5
  |
5 |     let (four, _one, _two) = try_indices!(&mut slice, 4, 1, 2).unwrap();
  |                                           ---------- `slice` is borrowed here
6 |     slice[4] *= 2;
  |     ^^^^^^^^ use of borrowed `slice`
7 |     *four *= 2;
  |     ---------- borrow later used here

error[E0503]: cannot use `slice[_]` because it was mutably borrowed
 --> tests/trybuild/try_indices/cannot_mutate_once_borrowed.rs:6:5
  |
5 |     let (four, _one, _two) = try_indices!(&mut slice, 4, 1, 2).unwrap();
  |                                           ---------- `slice` is borrowed here
6 |     slice[4] *= 2;
  |     ^^^^^^^^^^^^^ use of borrowed `slice`
7 |     *four *= 2;
  |     ---------- borrow later used here
